src/coreclr/scripts/superpmi.py

```diff
@@ -1438,68 +1437,62 @@ def replay_with_asm_diffs(self):
+                flags = [
+                    "-a",  # Asm diffs
+                    "-v", "ewmi",  # display errors, warnings, missing, jit info
+                    "-f", fail_mcl_file,  # Failing mc List
+                    "-diffMCList", diff_mcl_file,  # Create all of the diffs in an mcl file
+                    "-r", os.path.join(temp_location, "repro"),  # Repro name, create .mc repro files
+                    "-baseMetricsSummary", base_metrics_summary_file, # Create summary of metrics we can use to get total code size impact
+                    "-diffMetricsSummary", diff_metrics_summary_file,
+                ]
+                flags += altjit_asm_diffs_flags
+                flags += base_option_flags
+                flags += diff_option_flags
 
+                if not self.coreclr_args.sequential:
+                    flags += [ "-p" ]
+
+                if self.coreclr_args.break_on_assert:
+                    flags += [ "-boa" ]
+
+                if self.coreclr_args.break_on_error:
+                    flags += [ "-boe" ]
+
+                if self.coreclr_args.spmi_log_file is not None:
+                    flags += [ "-w", self.coreclr_args.spmi_log_file ]
+
+                if self.coreclr_args.error_limit is not None:
+                    flags += ["-failureLimit", self.coreclr_args.error_limit]
+
+                # Change the working directory to the Core_Root we will call SuperPMI from.
+                # This is done to allow libcoredistools to be loaded correctly on unix
+                # as the loadlibrary path will be relative to the current directory.
+                with ChangeDir(self.coreclr_args.core_root):
+                    command = [self.superpmi_path] + flags + [self.base_jit_path, self.diff_jit_path, mch_file]
+                    return_code = run_and_log(command)
+
+                base_metrics = read_csv_metrics(base_metrics_summary_file)
+                diff_metrics = read_csv_metrics(diff_metrics_summary_file)
+
+                print_superpmi_result(return_code, self.coreclr_args, base_metrics, diff_metrics)
                 artifacts_base_name = create_artifacts_base_name(self.coreclr_args, mch_file)
 
-                if is_nonzero_length_file(fail_mcl_file):
-                    # Unclean replay. Examine the contents of the fail.mcl file to dig into failures.
-                    if return_code == 0:
-                        logging.warning("Warning: SuperPMI returned a zero exit code, but generated a non-zero-sized mcl file")
-                    print_fail_mcl_file_method_numbers(fail_mcl_file)
-                    repro_base_command_line = "{} {} {}".format(self.superpmi_path, " ".join(altjit_asm_diffs_flags), self.diff_jit_path)
-                    save_repro_mc_files(temp_location, self.coreclr_args, artifacts_base_name, repro_base_command_line)
+                if return_code != 0:
+                    result = False
```

Comment:
Don't you need to also move this `result = False` within the `if return_code != 2 and return_code != 3:` condition?

Reply:
Oh yes.